pkg/segment/query/processor/searcher.go

```diff
@@ -465,6 +510,75 @@ func (s *Searcher) handleSortIndexWithFilter(qsr *query.QuerySegmentRequest, lin
 	return iqr, nil
 }
 
+func (s *Searcher) applyPQSForSortedIndex(qsr *query.QuerySegmentRequest, searchResults *segresults.SearchResults,
+	pqmrResults *pqmr.SegmentPQMRResults, searchMetadata map[uint16]*structs.BlockMetadataHolder, blkSummaries []*structs.BlockSummary,
+	sizeLimit uint64, aggs *structs.QueryAggregators, blockToValidRecNums map[uint16][]uint16) error {
+
+	if len(searchMetadata) == 0 {
+		log.Infof("qid=%d, applyPQSForSortedIndex: segKey %+v has 0 blocks in segment PQMR results", s.qid, qsr.GetSegKey())
+		return nil
+	}
+
+	// Remove blockNums that are not required to be processed.
+	searchMetadataToSend := make(map[uint16]*structs.BlockMetadataHolder)
+	validBlkNums := []uint16{}
+	for blockNum := range searchMetadata {
+		if _, ok := blockToValidRecNums[blockNum]; ok {
+			searchMetadataToSend[blockNum] = searchMetadata[blockNum]
+			validBlkNums = append(validBlkNums, blockNum)
+		}
+	}
+	pqmrToSend := pqmrResults.GetCopyOfBlockResults(validBlkNums)
+
+	req := &structs.SegmentSearchRequest{
+		SegmentKey:          qsr.GetSegKey(),
+		VirtualTableName:    qsr.GetTableName(),
+		AllPossibleColumns:  s.queryInfo.GetColsToSearch(),
+		AllBlocksToSearch:   searchMetadataToSend,
+		BlockToValidRecNums: blockToValidRecNums,
+		SearchMetadata: &structs.SearchMetadataHolder{
+			BlockSummaries:    blkSummaries,
+			SearchTotalMemory: uint64(len(blkSummaries) * 16), // TODO: add bitset size here
```

Comment:
1. I'm not sure what this TODO means
2. Should this 16 be 24 instead? The size of one BlockSummary is 8 + 8 + 2 + 6 (padding) = 24
```
type BlockSummary struct {
	HighTs   uint64
	LowTs    uint64
	RecCount uint16
}
```

Reply:
Sorry forgot to remove the TODO. Addressed in the latest commit.